Update package_item to follow DAT-549

kjoshi opened the pull request:
Various small changes to package_item so that the design matches what was requested in [DAT-549](https://london.atlassian.net/browse/DAT-549).

- Use the `search_description` if it's available, otherwise fall back to the truncated main description.
- Move `last updated` time to the top.
- Add "hosted by <harvest source>" under the dataset title, using the `harvest_source_title`.
- Add the dataset `entry_type` (default to `Dataset` if not present) and add the available resource formats.

The only issue was the `harvest_source_title`, which for some reason wasn't always being extracted from the package.extras into its own key. I've come up with a hacky workaround using jinja2's namespaces (see the `Scoping Behaviour` section in the [Assignments](https://jinja.palletsprojects.com/en/2.11.x/templates/#assignments) section of the docs) and looping over the extras, in case the convert_from_extras doesn't work when deployed.

Commit:
Update package_item to follow DAT-549

Various small changes to package_item so that the design matches what
was requested in DAT-549.

The only issue was the harvest_source_title, which for some reason
wasn't being extracted from the package.extras into its own key.
I've come up with a hacky workaround using jinja2's namespaces and
looping over the extras, in case the convert_from_extras doesn't work
when deployed.

madelinekosse reviewed (changes requested):
Maybe this isn't a significant enough problem to deal with right now, but the updated date text is misaligned:

![image](https://github.com/GreaterLondonAuthority/dfl-ckanext/assets/20282373/56edf012-bfc4-44a3-933b-cf84e1055a65)

Appreciate that Sven said not to worry about CSS too much, but adding a `width` to the updated date divs should be an easy fix

Commit:
Update 'last updated' to have constant width

On narrower screens the 'Last Updated' div was getting squashed and had
a different width depending on the contents. Fixed by not allowing that
flexbox to shrink.
(See https://www.w3.org/TR/css-flexbox-1/#flex-property)

kjoshi commented:
Thanks for checking! Just fixed that, I think it looks Ok now:
![image](https://github.com/GreaterLondonAuthority/dfl-ckanext/assets/1217533/c50b2500-b4c8-4788-9eca-d2e2dbc0dda7)

madelinekosse reviewed (approved):
Looks great, thanks!
